File: trunk/PO210640/task_04/Lab_04/Lab_04/Window.cs
namespace Lab_04;

public class Window
{
    #region Fields

    private static int _nextId = 0;

    protected int _id;
    protected int _width;
    protected int _height;
    protected int _memoryNeeded;
    protected bool _areAdministatorRightsGranted;
    protected bool _isShown;

    #endregion


    public Window(int width, int height, bool areAdministatorRightsGranted)
    {
        _id = _nextId++;

        _width = width;
        _height = height;
        _areAdministatorRightsGranted = areAdministatorRightsGranted;
    }

    protected virtual void DoWork()
    {
        // do smth
    }

    public void ShowOrHide(bool isShown)
    {
        _isShown = isShown;
    }
}

File: trunk/210647/task_04/src/Window.cpp
#include "Window.h"

Window::Window(int id, int height, int width, bool areAdministratorRightsGranted)//1//210647
    : id(id), height(height), width(width), areAdministratorRightsGranted(areAdministratorRightsGranted) {//1//210647
}

void Window::doWork() {//1//210647
    memoryNeeded += 10;//1//210647
}

void Window::showOrHide() {//1//210647
    isShown = false;//1//210647
}